qiskit_dynamics/solvers/solver_utils.py

```diff
@@ -116,17 +115,65 @@ def trim_t_results(
     if t_eval is None:
         return results
 
-    t_span = Array(t_span, backend="numpy")
+    # remove endpoints
+    results.t = results.t[1:-1]
+    results.y = Array(results.y[1:-1])
+
+    return results
+
+
+def merge_t_args_jax(t_span, t_eval=None):
+    """JAX-compilable version of merge_t_args. Rather than raise errors, sets return values to
+    jnp.nan to signal errors.
+    """
+
+    if t_eval is None:
+        return Array(t_span, backend="jax")
 
-    # remove endpoints if not included in t_eval
-    if t_eval[0] != t_span[0]:
-        results.t = results.t[1:]
-        results.y = Array(results.y[1:])
+    t_span = Array(t_span, backend="jax").data
+    t_eval = Array(t_eval, backend="jax").data
 
-    if t_eval[-1] != t_span[1]:
-        results.t = results.t[:-1]
-        results.y = Array(results.y[:-1])
+    out = jnp.append(jnp.append(t_span[0], t_eval), t_span[1])
+
+    # raise error if not one dimensional
+    if t_eval.ndim > 1:
+        raise ValueError("t_eval must be 1 dimensional.")
+
+    # output nan if t_eval point lies outside t_span
+    t_min = jnp.min(t_span)
+    t_max = jnp.max(t_span)
+    out = cond(
+        (jnp.min(t_eval) < t_min) | (jnp.max(t_eval) > t_max),
+        lambda s: jnp.nan * s,
+        lambda s: s,
+        out,
+    )
+
+    # output nan if t_eval and t_span have incompatible orderings
```

Comment:
why output `nan` and not raise an error?

Reply:
The error checking needs to be compilable/differentiable, but the error depends on the _values_ of the arrays. There's no way of doing value-dependent error raising in a traceable way in JAX, so in cases like this they recommend outputing `nan` as a way of signaling an error took place.